dnf/util.py

```diff
@@ -264,9 +263,10 @@ def on_ac_power():
 
     """
     try:
-        ret = subprocess.call('/usr/bin/on_ac_power')
-        return not ret
-    except OSError:
+        with open("/sys/class/power_supply/AC/online") as ac_status:
+            data = ac_status.read()
+            return int(data) == 1
+    except (IOError, ValueError):
         return None
```

Comment:
We need to assume on AC power unless told otherwise. This logic will not do that if the sysfs location doesn't exist, will it?

Reply:
I believe that the implementation is is correct. Def on_ac_power() returns True (on AC), False (on battery), None (Unable to detect). Then usage in code is like:
`dnf.util.on_ac_power() is False:`
It means that when None is returned it assumes that it is not on battery.